dask_cuda/disk_io.py

```diff
@@ -164,12 +166,11 @@ def disk_write(path: str, frames: Iterable, shared_filesystem: bool, gds=False)
     cuda_frames = tuple(hasattr(f, "__cuda_array_interface__") for f in frames)
     frame_lengths = tuple(map(nbytes, frames))
     if gds and any(cuda_frames):
-        import cucim.clara.filesystem as cucim_fs
+        import kvikio
 
-        with cucim_fs.open(path, "w") as f:
+        with kvikio.CuFile(path, "w") as f:
             for frame, length in zip(frames, frame_lengths):
-                f.pwrite(buf=frame, count=length, file_offset=0, buf_offset=0)
-
+                f.pwrite(buf=frame, count=length, file_offset=0, buf_offset=0).get()
```

Comment:
Yeah originally tried this and think it may work. However was a little unsure whether this could introduce race conditions as there is not a good way to chain these currently. So decided to punt on that. Maybe this can be handled in a follow up?

Reply:
Do we need to chain them?

Reply:
They are writing to the same file

Reply:
That is support, as long as the IO doesn't overlap.
In fact, by default KvikIO reads and writes to the same file concurrently using a thread pool.

Reply:
Interesting thanks for the clarification! 🙏

Maybe we can follow up on this in a subsequent PR?

Switching to KvikIO in Dask-CUDA will simplify the CUDA 12 migration effort (as Dask-CUDA won't need to wait on cuCIM)

Reply:
Sure

Reply:
As follow up submitted PR ( https://github.com/rapidsai/dask-cuda/pull/1205 ). It's possible more work is still needed there